specification/columns/commitmentutilizationtotal.md

```diff
@@ -0,0 +1,51 @@
+# Commitment Utilization Total
+
+Commitment Utilization Total is the numeric value representing the number of [Commitment Units](#glossary:commitment-unit) eligible for a resource or service.
+
+CommitmentUtilizationTotal MUST be present in the billing data when the provider supports *commitment-based discounts*.
+
+Cloud-based commitment-based discounts, like Reservations and Savings Plans, are commonly allocated to hourly [Charge Periods](#glossary:chargeperiod) of a billing period. When a commitment-based discount is allocated by a fixed charge period granularity (i.e. hourly), the CommitmentUtilizationTotal value MUST be the total amount of commitment units eligible for that charge period.
```

Comment:
Should the total amount be present only for a single line item (maybe identified with ChargeCategory=CommitmentUtilization)?
Or should it be present for all line items where the commitment discount is applied?

Reply:
Note: This pr needs to be updated based on decisions that didn't land (i.e. `ChargeSubCategory`)

So, I only see 2 approaches here (open to more ideas though):

1. 1 row showing aggregate used/total utilization amounts.

Columns `CommitmentUtilizationTotal` and `CommitmentUtilizationUsed` exist on 1 row of `ChargeCategory:Usage, ChargeFrequency:Recurring` per granularity (ex: hourly)

Unused amount are derived as: `CommitmentUtilizationTotal - CommitmentUtilizationUsed`

--

2. N rows for `CommitmentUtilizationUsed` amounts on `ChargeCategory:Usage,CommitmentDiscountStatus:Used` rows, and 1 aggregate recurring row for *only* Unused totals.

Column `CommitmentUtilizationUsed` exists on N rows of `ChargeCategory:Usage, CommitmentUtilizationStatus:Used` representing the total amount used per resource/service

`CommitmentUtilizationTotal` exists 1 row of `ChargeCategory:Usage, ChargeFrequency:Recurring` representing the total amount allocated for the granularity. (like Option 1)

Unused amount derived as: `CommitmentUtilizationTotal - SUM(CommitmentUtilizationUsed)`

--

While option 2 explicitly shows to see how much of the hourly commitment total was allocated to a single resource, this can also be derived via Option 1.

I don't personally think Option 2 gives that much more benefit but open to other opinions.